libs/internals/core.write_compiled_resource.php

```diff
@@ -6,6 +6,45 @@
  */
 
 /**
+ * Check to see if the path is writable, works on Windows platforms
+ * Based from http://php.net/manual/en/function.is-writable.php comment by
+ * legolas558
+ *
+ * Added 04/03/2015 by Yola to address is_writable PHP bug that does not
+ * correctly handle ACL on Windows platforms.
+ */
+function isWritable($path)
+{
+    $isWin = strtoupper(substr(php_uname('s'), 0, 3)) === 'WIN';
+
+    if($isWin) {
+        if ($path{strlen($path) - 1} == '/') {
+            return isWritable($path.uniqid(mt_rand()) . '.tmp');
+        } elseif (is_dir($path)) {
+            return isWritable($path . '/' . uniqid(mt_rand()) . 'tmp');
+        }
+
+        // check tmp file for read/write capabilities
+        $rm = file_exists($path);
+        $f = @fopen($path, 'a');
+
+        if ($f === false) {
+            return false;
+        }
+
+        fclose($f);
+
+        if (!$rm) {
```

Comment:
I should add a comment, but it's actually "if fopen created the file, delete it". Avoids a highly unlikely name collision that results in an unrelated temp file being deleted when it shouldn't have been. It might be unnecessary precaution, though. Thoughts?